Sources/_CryptoExtras/RSA/RSA.swift

```diff
@@ -318,6 +319,65 @@ extension _RSA.Signing {
     }
 }
 
+extension _RSA.Encryption {
+    public typealias PublicKey = _RSA.Signing.PublicKey
+    public typealias PrivateKey = _RSA.Signing.PrivateKey
+}
+
+extension _RSA.Encryption {
+    public struct RSAEncryptedData: ContiguousBytes {
+        public var rawRepresentation: Data
+        
+        public init<D: DataProtocol>(rawRepresentation: D) {
+            self.rawRepresentation = Data(rawRepresentation)
+        }
+        
+        public func withUnsafeBytes<R>(_ body: (UnsafeRawBufferPointer) throws -> R) rethrows -> R {
+            try self.rawRepresentation.withUnsafeBytes(body)
+        }
+    }
+    
+    public struct RSADecryptedData: ContiguousBytes {
+        public var rawRepresentation: Data
+        
+        public init<D: DataProtocol>(rawRepresentation: D) {
+            self.rawRepresentation = Data(rawRepresentation)
+        }
+        
+        public func withUnsafeBytes<R>(_ body: (UnsafeRawBufferPointer) throws -> R) rethrows -> R {
+            try self.rawRepresentation.withUnsafeBytes(body)
+        }
+    }
+}
+
+extension _RSA.Encryption {
+    public struct Padding {
+        internal enum Backing {
+            case pkcs1_oaep
+        }
+        
+        internal var backing: Backing
+        
+        private init(_ backing: Backing) {
+            self.backing = backing
+        }
+        
+        public static let PKCS1_OAEP = Self(.pkcs1_oaep)
+    }
+}
+
+extension _RSA.Encryption.PrivateKey {
+    public func decrypt<D: DataProtocol>(_ data: D, padding: _RSA.Encryption.Padding) throws -> _RSA.Encryption.RSADecryptedData {
```

Comment:
As constructed, these encrypt/decrypt functions require messages to be of limited size. I'm fine with that as a constraint, but we should make it clear. Can you document the constraint on these functions?

Reply:
Ah, right... pardon my mental dust; it's been awhile since I dug into crypto 😅